### DISCUSSION TITLE
Custom gitlab server using git+ssh and http for api and web

### DISCUSSION THREAD

**Discussion by ldeck** [magit/forge#358] [Support] [answered]
I've added to ~/.authinfo
```
machine gitlab.internal:80/api/v4 login ldeck^forge password TOKEN
```

And appended to the forge-alist (not sure if this is right):
```
(add-to-list 'forge-alist
    '("gitlab.internal" "gitlab.internal:80/api/v4" "gitlab.internal" forge-gitlab-repository))
```

Added to ~/.gitconfig
```
[gitlab "gitlab.internal:80/api/v4"]
        user = ldeck
```

What do I need to do to have forge use http for api requests?

I get the following error in *Messages* when attempting to fetch all topics, for example,
```
gnutls.el: (err=[-15] An unexpected TLS packet was received.) boot: (:priority NORMAL:-VERS-TLS1.3 :hostname gitlab.internal :loglevel 0 :min-prime-bits nil :trustfiles (/etc/ssl/cert.pem) :crlfiles nil :keylist nil :verify-flags nil :verify-error nil :callbacks nil)
gnutls-negotiate: GnuTLS error: #<process gitlab.internal>, -15
```

It's clearly trying to use https instead of http.

The magit remote is `git@gitlab.internal:Path/To/repo.git`
git will use ssh, but forge's usage of the api needs to http.

**Comment by tarsius** [comment, marked as answer]
You'll also need `(add-to-list 'ghub-insecure-hosts "gitlab.internal:80/api/v4")`.

**Reply by ldeck** [reply]
Perfect. Thanks @tarsius !
